# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/ipc/CrossProcessPaint.h
# Commit: c2741585119b
# Full Hash: c2741585119ba8536918b94429296479c0865c25
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2021-07-27 09:45:43
# Regressor Bug: 1695806
# File Overlap Count: 1
# Description:
#   Bug 1695806 - Defer sending a snapshot reqeust for remote process iframe until cloning the iframe document finished. r=mattwoodrow,nika
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119950
# ==============================================================================

diff -r 3c5186df3677 -r c2741585119b gfx/ipc/CrossProcessPaint.h
--- a/gfx/ipc/CrossProcessPaint.h	Tue Jul 27 00:47:44 2021 +0000
+++ b/gfx/ipc/CrossProcessPaint.h	Tue Jul 27 00:47:44 2021 +0000
@@ -30,6 +30,7 @@
 namespace mozilla {
 
 namespace dom {
+class CanonicalBrowsingContext;
 class DOMRect;
 class Promise;
 class WindowGlobalParent;
@@ -145,6 +146,8 @@
       nscolor aBackgroundColor = NS_RGBA(0, 0, 0, 0),
       CrossProcessPaintFlags aFlags = CrossProcessPaintFlags::DrawView);
 
+  void QueuePaint(dom::CanonicalBrowsingContext* aBc);
+
   /// Clear the state of this paint so that it cannot be resolved or receive
   /// any paint fragments.
   void Clear(nsresult aStatus);